src/WebJobs.Script.WebHost/Management/LegionInstanceManager.cs

```diff
@@ -0,0 +1,146 @@
+        }
+
+        public Task<string> SpecializeMSISidecar(HostAssignmentContext context)
+        {
+            // Legion will take care of MSI Specialization
+            return null;
+        }
+
+        public bool StartAssignment(HostAssignmentContext context)
+        {
+            if (!_webHostEnvironment.InStandbyMode && !context.Environment.TryGetValue(EnvironmentSettingNames.ContainerStartContext, out string startContext))
+            {
+                // This is only true when specializing pinned containers.
+                _logger.LogError("Assign called while host is not in placeholder mode and start context is not present.");
+                return false;
+            }
+
+            if (_environment.IsContainerReady())
+            {
+                _logger.LogError("Assign called while container is marked as specialized.");
+                return false;
+            }
+
+            if (context.IsWarmupRequest)
+            {
+                // No need to download since file already exists
+                return true;
+            }
+
+            if (_assignmentContext == null)
+            {
+                lock (_assignmentLock)
+                {
+                    if (_assignmentContext != null)
+                    {
+                        return _assignmentContext.Equals(context);
+                    }
+                    _assignmentContext = context;
+                }
+
+                _logger.LogInformation($"Starting Assignment. Cloud Name: {_environment.GetCloudName()}");
+
+                // set a flag which will cause any incoming http requests to buffer
+                // until specialization is complete
+                // the host is guaranteed not to receive any requests until AFTER assign
+                // has been initiated, so setting this flag here is sufficient to ensure
+                // that any subsequent incoming requests while the assign is in progress
+                // will be delayed until complete
+                _webHostEnvironment.DelayRequests();
+
+                // start the specialization process in the background
+                Task.Run(async () => await Assign(context));
+
+                return true;
+            }
+
+            // No lock needed here since _assignmentContext is not null when we are here
+            return _assignmentContext.Equals(context);
+        }
+
+        public Task<string> ValidateContext(HostAssignmentContext assignmentContext)
+        {
+            // Don't need to validate RunFromPackageContext in Legion
+            return null;
+        }
+
+        private async Task Assign(HostAssignmentContext assignmentContext)
+        {
+            try
+            {
+                // first make all environment and file system changes required for
+                // the host to be specialized
+                _logger.LogInformation("Applying {count} app setting(s)", assignmentContext.Environment.Count);
```

Comment:
nit: `{environmentCount}` - `{count}` is not very clear _what_ is the subject of the count.

Reply:
Fixed!